More detailed setup for blend on other distros (#7)
blend-os/blend: README.md

## Usage

It's recommended to use the `blend-settings` UI, instead of the `blend` CLI.

> dependentcies require for both methods
> 
> podman 

Reviewer: ```suggestion
Both methods require `podman` as a dependency.
```

> ## Building Manually

Reviewer: ```suggestion
## Building Manually

```
It is recommended to leave a blank line space after a heading in Markdown.

> To build manually, you are going to need the following dependentcies

Reviewer: ```suggestion
To build manually, you are going to need the following dependencies:
```

> pacman fakeroot git binutils npm electron(maunaully downloaded from github, see later) bzip2 make gcc

Reviewer: ```suggestion
`pacman fakeroot git binutils npm electron bzip2 make gcc`
```

Author: The PKG build looks specifically for a folder named electron 22, when the current version is 24

> Install the dependentcies from your package manager except for electron
> 
> i.e sudo dnf install pacman fakeroot git binutils npm bzip2 make gcc

Reviewer: ```suggestion
Install the dependencies from your package manager (except electron) using the following command:

`sudo dnf install pacman fakeroot git binutils npm bzip2 make gcc`
```

> Clone the repo for building blend
> 
> git clone https://github.com/blend-os/blend-pkg

Reviewer: ```suggestion
Clone the repo for building blend:

`git clone https://github.com/blend-os/blend-pkg`
```

> Then download electron from https://github.com/electron/electron/releases/download/v24.2.0/electron-v24.2.0-linux-x64.zip

Reviewer: ```suggestion
Download electron from https://github.com/electron/electron/releases/latest
```
It is recommended to use latest, as the link will get outdated with newer releases.

> 
> Extract the zip and rename it to electron22
> 
> Go to where you downloaded it
> 
> cd Downloads
> 
> Copy it to /usr/lib (on mutable systems)
> 
> sudo cp electron22 -R /usr/lib
> 
> On immutable systems, you will need to edit the pkgbuild in blend-pkg to set a different directory
> 
> Go to blend-pkg
> 
> cd blend-pkg
> 
> Edit the file with editor of choice
> 
> (text editor) PKGBUILD
> 
> Edit this line to a directory you can copy to
> 
> electronVer="$(sed s/^v// /usr/lib/electron${_electronversion}/version)"
> 
> i.e on silverblue /var/usrlocal/lib/electron
> 
> Go back to home
> 
> cd 
> 
> Go to where you downloaded it
> 
> cd Downloads
> 
> Then copy to chosen directory
> 
> sudo cp electron22 -R /var/usrlocal
> 
> Go back to home
> 
> cd
> 
> Go to the blend-pkg folder
> 
> cd blend-pkg
> 
> This is where the guide splits from Arch to non Arch

Reviewer: ```suggestion

Extract the zip and rename it to electron22

Navigate to Downloads:

`cd Downloads`

Copy it to /usr/lib (on mutable systems):

`sudo cp electron22 -R /usr/lib`

On immutable systems, you will need to edit the pkgbuild in blend-pkg to set a different directory

Go to blend-pkg directory:

`cd blend-pkg`

Edit the file with the editor of choice:

`<text editor> PKGBUILD`

Edit the line to your correct directory:

`electronVer="$(sed s/^v// /usr/lib/electron${_electronversion}/version)"`

i.e on silverblue /var/usrlocal/lib/electron

Navigate back to home:

`cd  && cd Downloads`

Then copy the content to the selected directory

`sudo cp electron22 -R /var/usrlocal`

Navigate back to the blend-pkg folder:

`cd && cd blend-pkg`
```

> ##Arch

Reviewer: ```suggestion
### Installing blend in Arch
```

> Install podman from your distro's repo.
> 
> Proceed building the package and installing it and any missing dependentcies(if you forgot) with
> 
> makepkg -si
> 
> The reason why this is the arch version is that -s uses pacman to install dependetcies, which on non arch wouldnt work, and -i would install as a Arch app
> 
> may have unintended consequences on non arch.
> 
> run the /etc/profile.d/blend.sh  to set the path
> 
> Go to location
> 
> cd /etc/profile.d/blend.sh 
> 
> Set the sh as executable
> 
> chmod +x blend.sh
> 
> Run the script
> 
> ./blend.sh
> 
> Enable the blend-files service
> 
> cd /usr/lib/systemd/user
> 
> systemctl --user enable blend-files.service
> 
> #Done

Reviewer: ```suggestion
Install `podman`:

`sudo pacman -S podman`

Build the package:

`makepkg -si`

Execute `/etc/profile.d/blend.sh`  to set the path.

Navigate to the following location:

`cd /etc/profile.d/blend.sh` 

Make the script executable:

`chmod +x blend.sh`

Run the script:

`./blend.sh`

Enable the blend-files service:

`cd /usr/lib/systemd/user`

`systemctl --user enable blend-files.service`

```

copy all the contents of the extracted tar files to workspaceBlend; merge the folders and skip the replace(dont need buildinfo,mtree or pkginfo)

(so each extracted tar has its own usr directory, so copying that to workpaceBlend will merge them)

Here comes another split

On mutable distros,copy the usr folder to /usr and the etc folder to /etc; merge the folders

Enter workspaceBlend

cd workspaceBlend

sudo cp usr -R /usr

sudo cp etc -R /etc

On immutable distros, rename the /usr folder to what you need based on the overlay filesystem setup you have

On silverblue you would rename to usrlocal and copy to /var/usrlocal; merge the folders

sudo cp usrlocal -R /var/usrlocal

Do the same with /etc if you need

You dont need to on Silverblue though so proceed as normal.

sudo cp etc -R /etc

run the /etc/profile.d/blend.sh  to set the path

Go to location

cd /etc/profile.d/blend.sh 

Set the sh as executable

chmod +x blend.sh

Run the script

./blend.sh

Enable the blend-files service

cd /usr/lib/systemd/user

systemctl --user enable blend-files.service

Thats the main Install, some goodies below

To start on log in add the following to .bash_profile

blend start-containers

> 
> 
> 
> 
>              
> 
> 
> 
> 
> 

Reviewer: ```suggestion
```
